### PR TITLE
PCHR-2430: Fix leave requests action labels

### PR DESCRIPTION
**Pull request by igorpavlov**
## Overview

The Leave Request Action Matrix has been updated. This PR makes the leave request actions comply with it.

## Technical Details

The most major change is amending the following function:

```js
function canLeaveRequestBeCancelled (leaveRequestStatus) {
  var allowCancellationValue = vm.absenceTypes[vm.leaveRequest.type_id]
    .allow_request_cancelation;

  // Admin can always cancel
  if (vm.role === 'admin') {
    return true;
  }

  // Others but Admin can cancel if only the leave request status is
  // either "Awaiting for Approval" or "More Information Required"
  if (!_.includes([statusNames.awaitingApproval, statusNames.moreInformationRequired],
    leaveRequestStatus)) {
    return false;
  }

  // Manager can cancel disregarding the allow_request_cancelation value
  if (vm.role === 'manager') {
    return true;
  }

  // If request can only be cancelled in advance of start date
  if (allowCancellationValue === '3') {
    return moment().isBefore(vm.leaveRequest.from_date);
  }

  // If request can always be cancelled
  return allowCancellationValue === '2';
}
```

## Comments

This PR also fixes the issue with the Filter by Assignee in the Admin Leave Requests. The issue was not with the component itself, but in its HTML template, where the wrong value was set on click. You can notice in the Before GIF the "All" is not selected, but in the After GIF it is.

BackstopJS tests are added because the Filter by Assignee is a specific feature for Admin only.

---

- [x] Tests Pass

### CODE REVIEW

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-actions.component.js [hunk 1]
```diff
@@ -138,29 +150,40 @@ define([
...
-     * 3 = REQUEST_CANCELATION_IN_ADVANCE_OF_START_DATE
+     * @TODO This function utilises external resource
+     * vm.absenceTypes - this sould be refactored
      *
      * @return {Boolean}
      */
-    function canLeaveRequestBeCancelled () {
-      var allowCancellationValue = vm.absenceTypes[vm.leaveRequest.type_id].allow_request_cancelation;
+    function canLeaveRequestBeCancelled (leaveRequestStatus) {
+      var allowCancellationValue = vm.absenceTypes[vm.leaveRequest.type_id]
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-actions.component.js]: Please keep the `.` in the original(previous) line. Readability is better that way.

File: uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/manage-leave-requests.html [hunk 2]
```diff
@@ -19,7 +19,7 @@ <h3 class="col-md-5 panel-title" ng-if="!vm.isAdmin">Manager Absence Approval</h
             <button
               ng-repeat="filter in vm.filtersByAssignee"
               type="button"
-              ng-click="vm.refreshWithFilterByAssignee(filter.type)"
+              ng-click="vm.refreshWithFilterByAssignee(filter)"
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/manage-leave-requests.html]: The function expects a `String`, but you are sending object. Please update the document of the function.

**Comment by deb1990** [comment]
The syntax of commit messages are wrong
It should be "<TicketNO>: <Change Description>"

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-actions.component.js (hunk 1 above)
Comment (igorpavlov) [reply uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-actions.component.js]: Done.

**Pull request merged**